pkg/client/apiutil/apimachinery.go

```diff
@@ -49,6 +56,29 @@ func GVKForObject(obj runtime.Object, scheme *runtime.Scheme) (schema.GroupVersi
 // RESTClientForGVK constructs a new rest.Interface capable of accessing the resource associated
 // with the given GroupVersionKind.
 func RESTClientForGVK(gvk schema.GroupVersionKind, baseConfig *rest.Config, codecs serializer.CodecFactory) (rest.Interface, error) {
+	cfg := createRestConfig(gvk, baseConfig)
+	cfg.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: codecs}
+	return rest.RESTClientFor(cfg)
+}
+
+// RESTUnstructuredClientForGVK constructs a new rest.Interface for accessing unstructured resources.
+func RESTUnstructuredClientForGVK(gvk schema.GroupVersionKind, baseConfig *rest.Config) (rest.Interface, error) {
+	cfg := createRestConfig(gvk, baseConfig)
+	var jsonInfo runtime.SerializerInfo
+	for _, info := range scheme.Codecs.SupportedMediaTypes() {
+		if info.MediaType == runtime.ContentTypeJSON {
+			jsonInfo = info
+			break
+		}
+	}
+	jsonInfo.Serializer = dynamicCodec{}
```

Comment:
comment below about dynamicCodec doesn't boost my confidence. I thought initially it will be used in test code or so, but I see it is being used here in *real* code.

Reply:
Ack.